.github/CONTRIBUTING.md

```diff
@@ -69,102 +67,142 @@ There are few ways you can play with the build:
 
--  [Running automated tests](#running-automated-tests)
--  [Performing static code analysis](#static-code-analysis)
+For features, use test page as a playground. Write new tests to verify different sub-features, e.g. rendering in carousel layout vs. rendering in stacked layout. Also write tests for both happy and unhappy paths. This will future-proof the work and protect the investment.
 
-## Running automated tests
+### Writing first test
 
-Automated tests in Web Chat are designed to run inside a stable and isolated test environment. To run the test suite, you will need to [start the test environment](#starting-the-test-environment), then [run the test suite](#running-the-test-suite).
+Start by copying the test page template from [`/__tests__/html/simple.html`](https://github.com/microsoft/BotFramework-WebChat/blob/main/__tests__/html/simple.html) and [`simple.js`](https://github.com/microsoft/BotFramework-WebChat/blob/main/__tests__/html/simple.js). Additionally, follow [other test pages](https://github.com/microsoft/BotFramework-WebChat/tree/main/__tests__/html) to learn about our [page object model](https://github.com/microsoft/BotFramework-WebChat/tree/main/packages/test/page-object/src/globals).
 
-### Starting the test environment
+We prefer using end-to-end visual regression tests (VRT) for pixel-perfect and whole feature verification. And unit tests for utility functions. For VRT, we use [`pixelmatch`](https://npmjs.com/package/pixelmatch) via [`jest-image-snapshot`](https://npmjs.com/package/jest-image-snapshot).
 
-For environment stability, Web Chat uses Docker for hosting the test environment.
+### Running tests manually
 
--  Docker with [Windows Subsystem for Linux 2 (a.k.a. WSL2)](https://aka.ms/wsl2)
-   -  To start, run `npm run start:docker`
-   -  To stop/cleanup, run `npm run stop:docker`
--  Other Docker setup
-   -  To start, run `docker-compose up --build`
-   -  To stop/cleanup, run `docker-compose down`
+Download [ChromeDriver](https://sites.google.com/a/chromium.org/chromedriver/downloads) and extract to the project root.
 
-> On Windows, set environment variable `COMPOSE_CONVERT_WINDOWS_PATHS=1`.
+Run `npm run browser`. It will open a new browser window to http://localhost:5001/**tests**/html/. Then, navigate to the test file.
 
-### Running the test suite
+When tests have completed successfully, it should show a green check.
 
-> Testing pages under `__tests__/html/` are served under http://localhost:5001/ (locally) and http://localhost:5081/ (from Docker). All Jest tests will run against Docker version of the page. To update the page, restart Docker.
+![Transcript with a green check showing test succeeded](https://github.com/microsoft/BotFramework-WebChat/raw/main/docs/media/running-test.png)
 
-There are 2 ways to run the test suite, either one-off or continuously:
+> If the test pages take any screenshots, run them in an automated manner to save the screenshots.
```

Comment:
'automated manner' is vague. How is this?:

'Run the new tests in Docker to save screenshots (see below). Verify that these screenshots match the behavior you intend to test.'

Reply:
Is this in reference to running one-off vs continuously? In that case, the old verbiage is clearer.

Reply:
No this is not related to one-off or continuously.

All of them are continuously now:

- `npm run browser`: This will run a browser only, but the end-dev should be running `npm start` in another terminal
- `npm run docker`: This will run Docker and rebuild/restart containers as needed (i.e. watch)
- `npm test`: This is Jest watch

Reply:
ok. How does top comment seem to you?

Reply:
Automated manner -> "run in Jest"?